File: angularapp/src/app/service/review-cycle.service.ts
// src/app/services/review-cycle.service.ts

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ReviewCycle } from '../model/review-cycle.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ReviewCycleService {
  private baseUrl = `${environment.apiUrl}/reviewCycles`;

  constructor(private http: HttpClient) {}

  // Get all review cycles
  getAllReviewCycles(): Observable<ReviewCycle[]> {
    return this.http.get<ReviewCycle[]>(`${this.baseUrl}`);
  }

  // Get review cycle by ID
  getReviewCycleById(id: number): Observable<ReviewCycle> {
    return this.http.get<ReviewCycle>(`${this.baseUrl}/${id}`);
  }

  // Create review cycle
  createReviewCycle(cycle: ReviewCycle): Observable<ReviewCycle> {
    return this.http.post<ReviewCycle>(`${this.baseUrl}`, cycle);
  }

  // Update review cycle
  updateReviewCycle(id: number, cycle: ReviewCycle): Observable<ReviewCycle> {
    return this.http.put<ReviewCycle>(`${this.baseUrl}/${id}`, cycle);
  }

  // Delete review cycle
  deleteReviewCycle(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: springapp/src/main/java/com/example/springapp/service/ReviewCycleService.java
package com.example.springapp.service;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import com.example.springapp.model.ReviewCycle;
import com.example.springapp.repository.ReviewCycleRepository;

@Service
public class ReviewCycleService {

    private final ReviewCycleRepository repo;

    public ReviewCycleService(ReviewCycleRepository repo) {
        this.repo = repo;
    }

    public List<ReviewCycle> getAllReviewCycles() {
        return repo.findAll();
    }

    public Optional<ReviewCycle> getReviewCycleById(Long id) {
        return repo.findById(id);
    }

    public ReviewCycle createReviewCycle(ReviewCycle cycle) {
        return repo.save(cycle);
    }

    public ReviewCycle updateReviewCycle(ReviewCycle cycle) {
        return repo.save(cycle);
    }

    public void deleteReviewCycle(Long id) {
        repo.deleteById(id);
    }
}
